### PR TITLE
[10.0][ADD] add module shopinvader_product_selector

### PR DESCRIPTION
**Pull request by sebastienbeau**
The aim of this module is to had additionnal data so it's easy in the front to add a basic multi-field selector to select the variant.

Example:
color [blue, red]
size [M, L, XL]

Everytime we select a field we go to the matching variant.

This solution have been designed with Thibault, it's really simple to integrate in front.

As this is not needed on every projet (maybe you do not have variant, or only one dimension of variant).
I prefers to have this code in a separated module

### CODE REVIEW

File: shopinvader_product_variant_selector/readme/DESCRIPTION.rst
Comment (simahawk) [inline shopinvader_product_variant_selector/readme/DESCRIPTION.rst]:
```suggestion
This module adds the field `variant_selector` in the product variant metadata (exported to algolia or elastic). 
The value of the field contains the possible combinations of variants that are available.
This ease frontend development of variant selectors.
```

File: shopinvader_product_variant_selector/__manifest__.py
Comment (simahawk) [inline shopinvader_product_variant_selector/__manifest__.py]: remove these

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py:24]: what about stock availability? Have you planned something?

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py]:
```suggestion
        """Return the first variant that matches the values."""
```

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py]: you don't have to pass  `False`

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py]: why would you retrieve ALL attributes? This recordset can be huuuuge. :thinking: 
Each product should deal only w/ its related attributes (the ones attached to the template).

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py]: *preferred
Also, why "preferred"? Are preferred in a specific way?

File: shopinvader_product_variant_selector/tests/test_shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/tests/test_shopinvader_variant.py]: please use `SavepointCase` + `setUpClass` + `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

File: shopinvader_product_variant_selector/tests/test_shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/tests/test_shopinvader_variant.py]: white 32 GB?

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (simahawk) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py:63]: FTR in v12 there are several helpers fields on the template to determine which variant is available:
https://github.com/odoo/odoo/blob/12.0/addons/product/models/product_template.py#L122
Maybe worth to link this in a TODO comment.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (sebastienbeau) [reply shopinvader_product_variant_selector/models/shopinvader_variant.py:24]: Not yet

File: shopinvader_product_variant_selector/tests/test_shopinvader_variant.py
Comment (sebastienbeau) [reply shopinvader_product_variant_selector/tests/test_shopinvader_variant.py]: The SKU is used to be able to select the right variant on the front. As this value is not available there is not sku (variant is not active so this variant do not exist on elastic/algolia)

**Comment by sebastienbeau** [comment]
@simahawk This PR should be ready, you can review it.
Regarding the migration I see that the demo data have change between 10 and 12.
To avoid useless divergence between the two version, I propose to create a new set of demo data in this module (and so in this PR). So we will have no difference between 10, 12, 13 .. 14?
What to you think?

**Comment by simahawk** [comment]
@sebastienbeau I'm always in favour of test data isolation: go for it.

**Review by simahawk — APPROVED** [review]

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (lmignon) [inline shopinvader_product_variant_selector/models/shopinvader_variant.py]: @sebastienbeau  what's the meaning of `<=` operator?

**Review by lmignon — CHANGES_REQUESTED** [review]

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
@sebastienbeau Don't merge before fixing travis!

**Comment by sebastienbeau** [comment]
@lmignon yes I am working on this. I am waiting for a green travis before trying to merge.

File: shopinvader_product_variant_selector/models/shopinvader_variant.py
Comment (sebastienbeau) [reply shopinvader_product_variant_selector/models/shopinvader_variant.py]: it's True if all records of "variant.attribute_value_ids" are include into values.

**Comment by sebastienbeau** [comment]
Travis is green !! Let's merge it !
/ocabot merge

**Pull request merged**
